File: 上机1/pro1.cpp
#include <iostream>
#include <math.h>
#include <iomanip>
#include "stdlib.h"
using namespace std;
int main()
{
    long double sum = 0;
    int i;
    for (i = 10000; i >= 1; i--)
    {
        double j = i;
        sum = sum + 1 / (j * j * j * j);
    }
    cout << setprecision(100) << sum << endl;
    long double result[4] = {1.082322100000, 1.082323200000, 1.08232323371086100000, 1.08232323371080500000};
    long double error[4];
    for (i = 0; i < 4; i++)
    {
        error[i] = abs((result[i] - sum) / sum);
        cout << setprecision(9) << error[i] << endl;
    }
}

File: 上机1/pro1.py
from decimal import *
from turtle import end_fill
getcontext().prec = 100
sum = Decimal('0')
for i in range(1, 101, 1):
    sum = sum+Decimal('1')/(Decimal(str(i))*Decimal(str(i))
                            * Decimal(str(i))*Decimal(str(i)))
result = [Decimal('1.0823221'), Decimal('1.0823232'), Decimal(
    '1.082323233710861'), Decimal('1.082323233710805')]
error = [0, 0, 0, 0]
for i in range(4):
    error[i] = abs((result[i]-sum)/sum)
print(sum)
print(error)
